foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiChatCompletionRequest.java

```diff
@@ -0,0 +1,252 @@
+package com.sap.ai.sdk.foundationmodels.openai;
+
+import com.google.common.annotations.Beta;
+import com.google.common.collect.Lists;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.ChatCompletionStreamOptions;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.ChatCompletionTool;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.ChatCompletionToolChoiceOption;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.CreateChatCompletionRequest;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.CreateChatCompletionRequestAllOfResponseFormat;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.CreateChatCompletionRequestAllOfStop;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Value;
+import lombok.With;
+import lombok.experimental.Tolerate;
+
+/**
+ * Represents a request for OpenAI chat completion, including conversation messages and parameters.
+ *
+ * @since 1.4.0
+ */
+@Beta
+@Value
+@With
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@Getter(value = AccessLevel.NONE)
+public class OpenAiChatCompletionRequest {
+  /** List of messages from the conversation. */
+  @Nonnull List<OpenAiMessage> messages;
+
+  /** Stop sequences for the completion. */
+  @Nullable List<String> stop;
+
+  /** Temperature for the completion. */
```

Comment:
Can we have better Javadoc in here?
An example of stop sequences, what they do.
What number we can put in for temperature?
What is the diff between `maxTokens` and `maxCompletionTokens`?
etc